### DISCUSSION TITLE
[Feature] Hide language input switcher

### DISCUSSION THREAD

**Discussion by MaxPrimeAERY** [valinet/ExplorerPatcher#2126] [Ideas]
Hello. Can you please a feature that hides this annoying window when you change language ? I know there are replacement for InputSwitch.dll on GitHub but I want to do that for current Windows 10 22H2 (Probably also for Windows 11) without rewriting system files.

This window appears when you hold ALT and press SHIFT (or Win+Space).
![image](https://github.com/valinet/ExplorerPatcher/assets/30773045/9e6b80d6-228f-411c-8b83-d77a2957c5e7)

**Comment by pyrates999** [comment]
Please post windows version and build.  You can find this by running winver.
Please post explorer patcher version.

**Comment by MaxPrimeAERY** [comment]
> Please post windows version and build. You can find this by running winver. Please post explorer patcher version.

@pyrates999 
Windows 10 Pro 22H2 (build OS 19045.3324)
ExplorerPatcher v22621.1992.56.1

**Comment by pyrates999** [comment]
So when you press alt+shift together, you don't want the language bar showing?

**Comment by MaxPrimeAERY** [comment]
Yes

**Comment by pyrates999** [comment]
Try this: https://superuser.com/questions/698037/can-i-disable-the-altshift-shortcut-to-change-language-in-windows-8-1-or-win

**Comment by MaxPrimeAERY** [comment]
No. It doesn't solve it. I still need to change language with alt+shift but I don't want to see that window(screenshot above). I've written that there are a working solution https://balyberdin.com/hey/files/utilities/InputSwitch.rar but it makes Windows Defender Offline non-usable.

**Comment by SexyEvilGenius** [comment]
Hey! I faced the same problem and solved it today (at least for Win 11) by decompiling InputSwitch.dll and finding related registry keys. Here is a reg file for disabling the popup: https://gist.github.com/SexyEvilGenius/af320388399fa17fa03d92553b082265
It will correctly disable the popup on Alt+Shift AND Win+Space combinations, without altering the language switch behavior. Change is applied immediately without restart, so you can easily experiment with the values if you don't want to disable the popup completely.

**Reply by KernelPanic042** [reply]
Doesn't work for me(
Windows 10 LTSC 21H2
